src/modules/35-connectors/components/CreateConnector/PdcConnector/components/UploadJSON.tsx

```diff
@@ -0,0 +1,112 @@
+ */
+
+import React, { useEffect, useState, useRef, DragEvent } from 'react'
+import { Icon, Text } from '@harness/uicore'
+import { useToaster } from '@common/exports'
+import { useStrings } from 'framework/strings'
+import type { uploadHostItem } from '../StepDetails/PdcDetails'
+
+import css from './UploadJSON.module.scss'
+
+interface UploadJSONInterface {
+  setJsonValue: (value: uploadHostItem[]) => void
+}
+
+const UploadJSON = ({ setJsonValue }: UploadJSONInterface) => {
+  const { getString } = useStrings()
+  const { showError } = useToaster()
+  const inputRef = useRef() as React.MutableRefObject<HTMLInputElement>
+  const [fileContent, setFileContent] = useState([] as uploadHostItem[])
+  const [fileName, setFileName] = useState<string>('')
+  const [dropHighlight, setDropHighlight] = useState(false)
+
+  useEffect(() => {
+    setJsonValue(fileContent)
+  }, [fileContent])
+
+  const handleFileUpload = async (file: File) => {
+    setFileName('')
+    try {
+      const fr = new FileReader()
+      fr.onload = () => {
+        try {
+          setFileContent(JSON.parse(fr.result as string))
+          setFileName(file.name)
+        } catch (e) {
+          showError(e.message)
+        }
+      }
+      fr.readAsText(file)
+    } catch (e) {
+      showError(getString('connectors.pdc.errorUploading'))
+    }
+  }
+
+  const preventDefaults = (e: DragEvent<HTMLDivElement>) => {
+    e.preventDefault()
+    e.stopPropagation()
+  }
+
+  return (
+    <div
+      className={`${css.uploadComponent} ${dropHighlight ? css.highlightedDrop : ''}`}
+      onClick={() => inputRef.current.click()}
+      onDragEnter={e => {
+        preventDefaults(e)
+        setDropHighlight(true)
+      }}
+      onDragOver={e => {
+        preventDefaults(e)
+        setDropHighlight(true)
+      }}
+      onDragLeave={e => {
+        preventDefaults(e)
+        setDropHighlight(false)
+      }}
+      onDrop={event => {
+        preventDefaults(event)
+        setDropHighlight(false)
+        if (event.dataTransfer.items) {
+          // Use DataTransferItemList interface to access the file(s)
+          for (let i = 0; i < event.dataTransfer.items.length; i++) {
+            // If dropped items aren't files, reject them
+            if (event.dataTransfer.items[i].kind === 'file') {
+              const file = event.dataTransfer.items[i].getAsFile()
+              handleFileUpload(file as any)
+            }
+          }
+        } else {
+          // Use DataTransfer interface to access the file(s)
+          for (let i = 0; i < event.dataTransfer.files.length; i++) {
+            handleFileUpload(event.dataTransfer.files[i])
+          }
+        }
+      }}
+    >
+      <input
+        type="file"
+        id="bulk"
+        name="bulk-upload"
+        style={{ display: 'none' }}
+        ref={inputRef}
+        onChange={event => handleFileUpload((event.target as any).files[0])}
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232306)
```suggestion
        onChange={event => handleFileUpload((event.target as unknown).files[0])}
```